crates/sound/src/audio_backends/pw_backend.rs

```diff
@@ -133,16 +157,26 @@ impl PwBackend {
         println!("pipewire backend running");
 
         let streams_param = vec![PCMParams::default(); NR_STREAMS];
+        let streams_states = vec![0; NR_STREAMS];
 
         Self {
             thread_loop,
             core,
+            context,
             stream_params: RwLock::new(streams_param),
+            stream_states: RwLock::new(streams_states),
+            stream_hash: RwLock::new(HashMap::with_capacity(NR_STREAMS)),
         }
     }
 }
 
 impl AudioBackend for PwBackend {
+    fn get_stream_state(&self, stream_id: u32) -> Result<u32> {
```

Comment:
Do we need to check if stream_id is a valid id? We should display a msg if the id is invalid I think.

Reply:
I don't think it's necessary, and there's already a check for states at vhu_sound.rs:L361